mmseg/models/backbones/twins.py

```diff
@@ -0,0 +1,909 @@
+                groups=embed_dim))
+        self.s = s
+
+    def forward(self, x, H, W):
+        B, N, C = x.shape
+        feat_token = x
+        cnn_feat = feat_token.transpose(1, 2).view(B, C, H, W)
+        if self.s == 1:
+            x = self.proj(cnn_feat) + cnn_feat
+        else:
+            x = self.proj(cnn_feat)
+        x = x.flatten(2).transpose(1, 2)
+        return x
+
+    def no_weight_decay(self):
+        return ['proj.%d.weight' % i for i in range(4)]
+
+
+@BACKBONES.register_module()
+class PCPVT(PyramidVisionTransformer):
+    """Add applicable positional encodings to CPVT. The implementation of our
+    first proposed architecture: Twins-PCPVT.
+
+    Use useful results from CPVT.
+
+    PEG and GAP. Therefore, cls token is no longer required. PEG is used to
+    encode the absolute position on the fly, which greatly affects the
+    performance when input resolution changes during the training (such as
+    segmentation, detection)
+
+    Args:
+        img_size (int | tuple): Input image size. Default: 224.
+        patch_size (int): The patch size. Default: 4.
+        in_chans (int): Number of input channels. Default: 3.
+        num_classes (int): Number of num_classes. Default: 1000
+        embed_dims (list): embedding dimension. Default: [64, 128, 256, 512].
+        num_heads (int): number of attention heads. Default: [1, 2, 4, 8].
+        mlp_ratios (int): ratio of mlp hidden dim to embedding dim.
+            Default: [4, 4, 4, 4].
+        qkv_bias (bool): enable bias for qkv if True. Default: False.
+        drop_rate (float): Probability of an element to be zeroed.
+            Default 0.
+        attn_drop_rate (float): The drop out rate for attention layer.
+            Default 0.0
+        drop_path_rate (float): stochastic depth rate. Default 0.0
+        norm_cfg (dict): Config dict for normalization layer.
+            Default: dict(type='LN')
+        depths (list): depths of each stage. Default [3, 4, 6, 3]
+        sr_ratios (list): kernel_size of conv in each Attn module in
+            Transformer encoder layer. Default: [8, 4, 2, 1].
+        block_cls (BaseModule): Transformer Encoder.
+            Default TransformerEncoderLayer
+        F4=False（bool): input features need slice.
+        extra_norm（bool): add extra norm. Default False.
+    """
+
+    def __init__(self,
+                 img_size=224,
+                 patch_size=4,
+                 in_chans=3,
+                 num_classes=1000,
+                 embed_dims=[64, 128, 256, 512],
+                 num_heads=[1, 2, 4, 8],
+                 mlp_ratios=[4, 4, 4, 4],
+                 qkv_bias=False,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 norm_cfg=dict(type='LN'),
+                 depths=[3, 4, 6, 3],
+                 sr_ratios=[8, 4, 2, 1],
+                 block_cls=TransformerEncoderLayer,
+                 F4=False,
```

Comment:
The name `F4` is coordinate in MMSegmentation. We may rename it to something like `slice_input_features`.